### PR TITLE
RANCHER-1374: Add final Jenkinsfile and Dockerfile for agent.

### PR DESCRIPTION
**Pull request by sergii-msn**
Create a tiny base Jenkins agent docker image with tools required for common operations on Jenkins, it should include:

AWS CLI
Terraform client
Helm client
Psql and pg_dump client
Kafka client
Kubectl
Curl
Git
Java (version according to required by Jenkins for remote.jar)

Create a repository in ECR to store this Jenkins agent base image
Create a pipeline to automate build and publish operation for future upgrades

### CODE REVIEW

File: scripts/docker/jenkinsAgent/Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+def dockerfilePath = './scripts/docker/jenkinsAgent/'
+
+
+ansiColor('xterm') {
+  node('') {
+    try {
+        stage ('[Git] Checkout source') {
+          checkout scmGit(
+              branches: [[name: '*/RANCHER-1374']],
```
Comment (yaroslavishchenko) [inline scripts/docker/jenkinsAgent/Jenkinsfile:47]: Please don't forget to put here expected branch :)

File: scripts/docker/jenkinsAgent/Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,72 @@
+#!groovy
+
+import org.folio.Constants
+import org.folio.models.FolioModule
```
Comment (yaroslavishchenko) [inline scripts/docker/jenkinsAgent/Jenkinsfile:4]: Following packages are not used within this code:

import org.folio.models.FolioModule
import org.folio.models.InstallRequestParams
import org.folio.models.OkapiTenant
import org.folio.models.RancherNamespace
import org.folio.rest_v2.Main
import org.folio.utilities.GitHubClient

File: scripts/docker/jenkinsAgent/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+USER root
+RUN apk update && \
+    apk add --no-cache \
+    aws-cli \
+    postgresql-client \
+    unzip
+# Install Terraform
+RUN curl -fsSL "https://releases.hashicorp.com/terraform/${terraform_ver}/terraform_${terraform_ver}_linux_amd64.zip" -o terraform.zip && \
+    unzip terraform.zip && \
+    mv terraform /usr/local/bin/ && \
```
Comment (yaroslavishchenko) [inline scripts/docker/jenkinsAgent/Dockerfile:21]: Just in case:
Could you please make sure we don't need to run 
**chmod +x ${TOOL}** here and below :)

**Review by yaroslavishchenko — APPROVED** [review]

File: scripts/docker/jenkinsAgent/Dockerfile (hunk 3 above)
Comment (sergii-msn) [reply scripts/docker/jenkinsAgent/Dockerfile:22]: I`ve added chmod +x

File: scripts/docker/jenkinsAgent/Jenkinsfile (hunk 2 above)
Comment (sergii-msn) [reply scripts/docker/jenkinsAgent/Jenkinsfile]: removed extra libs

File: scripts/docker/jenkinsAgent/Jenkinsfile (hunk 1 above)
Comment (sergii-msn) [reply scripts/docker/jenkinsAgent/Jenkinsfile]: changed to parameters variable

**Pull request merged**